File: Algorithm/Python/power_of_two.py
class Solution:
    def isPowerOfTwo(self, n):
        return not n&(n-1)


if __name__ == '__main__':
    sol = Solution()
    import random
    n = random.randint(1,20000000)
    
    print n,sol.isPowerOfTwo(n)
    

File: Algorithm/CPP/power_of_two.cpp
# include <stdio.h>
# include <iostream>
# include <vector>
# include <algorithm>
using namespace std;
class Solution {
public:
    bool isPowerOfTwo(int n) {
    	if (n<=0) return false;
    	return (!(n&(n-1))) ;        
    }
};
int  main()
{
	Solution sol;

	//nums = { -1 };
	cout<<sol.isPowerOfTwo(-2147483648);
	cin.get();
	return 0;
}
